packages/gatsby-theme-vtex/src/sdk/buyButton/useBuyButton.ts

```diff
@@ -42,13 +50,18 @@ export const useBuyButton = (sku: Maybe<SKU>, quantity: number) => {
       setLoading(true)
       const items = [orderFormItem]
 
-      await orderForm.addToCart(items)
+      await orderForm.addToCart([orderFormItem])
+
       sendPixelEvent({
         type: 'vtex:addToCart',
         data: {
           items,
         },
       })
+
+      if (options?.isOneClickBuy) {
+        window.location.href = '/checkout/'
```

Comment:
is it assured the pixelEvent will be processed before navigating ?

Reply:
In Store Framework we added a spinner of 500ms, just to make sure the had a feedback and events were triggered

https://github.com/vtex-apps/add-to-cart-button/pull/44/files
https://github.com/vtex-apps/add-to-cart-button/pull/45/files